test/gdal_utils.jl

```diff
@@ -0,0 +1,162 @@
+ds_small = GDAL.open("data/utmsmall.tif", GDAL.GA_ReadOnly)
+ds_point = GDAL.openex("data/point.geojson", GDAL.GDAL_OF_VECTOR, C_NULL, C_NULL, C_NULL)
+
+# GDALInfo
+# infooptionsnew checks if the options are valid
+@test_throws GDAL.GDALError GDAL.infooptionsnew(["-novalidoption"], C_NULL)
+options = GDAL.infooptionsnew(["-json"], C_NULL)
+infodict = JSON.parse(GDAL.info(ds_small, options))
+GDAL.infooptionsfree(options)
+@test infodict["size"] == [100, 100]
+
+
+# GDALTranslate
+# resample to a 5×5 ascii grid
+optvec = ["-of","AAIGrid","-r","cubic","-tr","1200","1200"]
+options = GDAL.translateoptionsnew(optvec, C_NULL)
+ds_tiny_asc = GDAL.translate("data/utmtiny.asc", ds_small, options, C_NULL)
+GDAL.translateoptionsfree(options)
+GDAL.close(ds_tiny_asc)
+@test readstring("data/utmtiny.asc") == """
+    ncols        5
+    nrows        5
+    xllcorner    440720.000000000000
+    yllcorner    3745320.000000000000
+    cellsize     1200.000000000000
+     128 126 161 185 193
+     171 164 175 206 205
+     127 148 177 205 209
+     93 114 164 172 181
+     83 101 140 128 122
+    """
+# also create a Byte GeoTIFF version for later use
+optvec = ["-r","cubic","-tr","1200","1200","-ot","Byte"]
+options = GDAL.translateoptionsnew(optvec, C_NULL)
+ds_tiny = GDAL.translate("data/utmtiny.tif", ds_small, options, C_NULL)
+GDAL.translateoptionsfree(options)
+
+tinydata = UInt8[
+    128 126 161 185 193;
+    171 164 175 206 205;
+    127 148 177 205 209;
+     93 114 164 172 181;
+     83 101 140 128 122]
+
+band = GDAL.getrasterband(ds_tiny, 1)
+read_data = zeros(tinydata)
+GDAL.rasterio(band, GDAL.GF_Read, 0, 0, 5, 5,
+              read_data, 5, 5, GDAL.GDT_Byte, 0, 0)
+@test read_data' == tinydata
+
+
+# GDALWarp
+
+# cannot reproject file
+# ERROR: LoadError: LoadError: GDALError (Failure, code 6):
+#        Unable to load PROJ.4 library (libproj.so), creation of OGRCoordinateTransformation failed.
+# when using
+# options = GDAL.warpappoptionsnew(["-t_srs","EPSG:4326"], C_NULL)
+# should we require PROJ.4 or not?
```

Comment:
our options here might be to (i) [statically link PROJ.4](https://lists.osgeo.org/pipermail/gdal-dev/2004-October/004332.html), or (ii) [set the environment path variable](http://gis.stackexchange.com/a/157274) (if the dll exists).

Possibly relevant: https://github.com/ContinuumIO/anaconda-issues/issues/263